core/src/main/java/jeeves/server/dispatchers/ServiceManager.java

```diff
@@ -571,6 +572,33 @@ public ServiceContext createServiceContext(String name, int defaultUserId){
         return context;
     }
 
+    /**
+     * Provide best-guess language code (covering special case for 'fra' and 'slk' locales).
+     * <p>
+     * Logic matches LanguageUtils from module services.
+     * </p>
+     *
+     * @param locale Locale used to obtain {@link Locale#getISO3Language()} three-letter abbreviation
+     * @return language code, or "?" for unknown
+     */
+    private static String getIso3langCode(Locale locale){
+        if( locale == null ){
+            return "?";
+        }
+        String iso3language = locale.getISO3Language();
+        String code;
+        if(iso3language.equalsIgnoreCase("fra")){
```

Comment:
See the issue for discussion, we cannot see LanguageUtils from this module

Reply:
So it is in the service module. I don't see why it cannot be moved to the core.   There is a IsoLanguagesMapper class in the core and I think it would probably be better if that class supported the conversions from T to B and vice versa.  But probably more work that required at this time.